### PR TITLE
Add support for WebGL2 texture formats

### PR DESCRIPTION
**Pull request by NicolasBuecher**
Update WebGL constants to enable support for WebGL2 texture formats.
[Discussed here.](http://www.html5gamedevs.com/topic/38584-support-webgl2-new-texture-types-formats-and-sized-internal-formats/?tab=comments#comment-220446)

### CODE REVIEW

File: src/Engine/babylon.engine.ts [hunk 1]
```diff
@@ -359,17 +359,54 @@
...
-        /** R */
+        /** RED */
+        public static readonly TEXTUREFORMAT_RED = 6;
         public static readonly TEXTUREFORMAT_R = 6;
         /** RG */
         public static readonly TEXTUREFORMAT_RG = 7;
-
-        /** UNSIGNED_INT */
+        /** RED_INTEGER */
+        public static readonly TEXTUREFORMAT_RED_INTEGER = 8;
```
Comment (sebavan) [inline src/Engine/babylon.engine.ts]: Why do we need to add the integer here and below if it is already included in the TEXTURETYPE ?

File: src/Engine/babylon.engine.ts [hunk 2]
```diff
@@ -7042,18 +7121,35 @@
...
-                    internalFormat = this._gl.RED;
-                    break;       
-                case Engine.TEXTUREFORMAT_RG:
-                    internalFormat = this._gl.RG;
-                    break;                                    
+            }
+
+            if (this._webGLVersion > 1) {
+                switch (format) {
+                    case Engine.TEXTUREFORMAT_RED_INTEGER:
```
Comment (sebavan) [inline src/Engine/babylon.engine.ts]: Is this not possible to deduce from TEXTUREFORMAT_RED and type TEXTURETYPE_INT and same below ?

Comment (NicolasBuecher) [reply src/Engine/babylon.engine.ts]: For this example, it is possible. But for some others like TEXTURETYPE_BYTE and TEXTUREFORMAT_RED. You need here to specify TEXTUREFORMAT_R_INTEGER.

**Comment by NicolasBuecher** [comment]
As discussed [here](http://www.html5gamedevs.com/topic/38584-support-webgl2-new-texture-types-formats-and-sized-internal-formats/?do=findComment&comment=220258), DK said "I do not want to expose internal format." so I didn't add an extra internalSized

**Comment by NicolasBuecher** [comment]
Sorry but I won't be able to do any updates before Tuesday !
No matter if you want to make some changes during this time.

**Comment by sebavan** [comment]
Agree but it feels strange to me to see the format and type mixed together. This is only the integer by the way, all the rest looks good :-)

**Comment by sebavan** [comment]
No problem, no rush on this, this is a major change and it is better to look at it twice :-)

**Comment by sebavan** [comment]
@deltakosh, this directly maps with Webgl2 (reference: https://webgl2fundamentals.org/webgl/lessons/webgl-data-textures.html) so I wonder how we could avoid the confusion with the _INTEGER format which is truly a fromat in webgl and not mapping to a type.

Maybe we should leave it like this as it is a strong webgl match ?

**Comment by deltakosh** [comment]
Well yes I think this PR is probably the best we can get providing we want to support back compat

**Comment by sebavan** [comment]
So all good for me once the documentation checked issue has been fixed:
image

**Comment by deltakosh** [comment]
Try to run npm install gulp@4.0.0

**Comment by deltakosh** [comment]
I had to revert the PR as a lot of visual tests where failing (you can run them directly from vscode in the debug tasks (run visual tests))

**Comment by NicolasBuecher** [comment]
Thanks for gulp !!
I'll take a look at that.

**Comment by deltakosh** [comment]
I updated the readme :)

**Comment by deltakosh** [comment]
Please start a new one;)

**Pull request merged**